### PR TITLE
[simulation] enhance command line arguments by using get_opt

### PR DESCRIPTION
**Pull request by Irving-cl**
This PR extracts the `get_opt` enhancement part of #6536.

Since no new arguments is added to virtual simulation platform and 
it still only has one argument, I didn't use `get_opt` for it.

### CODE REVIEW

File: examples/platforms/simulation/system.c [hunk 1]
```diff
@@ -67,52 +67,68 @@ void otSysInit(int aArgCount, char *aArgVector[])
 {
     char *   endptr;
     uint32_t speedUpFactor = 1;
-    int      argIndex      = 0;
+
+    struct option long_options[] = {
+        {"sleep-to-tx", no_argument, 0, 's'},
+        {"time-speed-up-factor", required_argument, 0, 't'},
```
Comment (bukepo) [inline examples/platforms/simulation/system.c:73]: consistent with posix app.
```suggestion
        {"time-speed", required_argument, 0, 't'},
```

File: examples/platforms/simulation/system.c [hunk 2]
```diff
@@ -67,52 +68,68 @@ void otSysInit(int aArgCount, char *aArgVector[])
 {
     char *   endptr;
     uint32_t speedUpFactor = 1;
-    int      argIndex      = 0;
+
+    struct option long_options[] = {
+        {"sleep-to-tx", no_argument, 0, 's'},
```
Comment (zhanglongxia) [inline examples/platforms/simulation/system.c:73]: How about define an enumeration to represent the argument return values, refer to https://github.com/openthread/openthread/blob/main/src/posix/main.c#L146.

File: examples/platforms/simulation/system.c [hunk 3]
```diff
@@ -67,52 +68,68 @@ void otSysInit(int aArgCount, char *aArgVector[])
...
+        case 't':
+            speedUpFactor = (uint32_t)strtol(optarg, endptr, 10);
+            if (*endptr != '\0' || speedUpFactor == 0)
+            {
+                fprintf(stderr, "Invalid value for TimerSpeedUpFactor: %s\n", optarg);
+                exit(EXIT_FAILURE);
+            }
+            break;
+        case '?':
+            fprintf(stderr, "Syntax:\n    %s [--sleep-to-tx] [--time-speed-up-factor=val] NodeId \n", aArgVector[0]);
```
Comment (zhanglongxia) [inline examples/platforms/simulation/system.c:106]: Both the line 106 and 116 are printing the help info. How about add an option `-h` to output the help info?

**Review by zhanglongxia — COMMENTED** [review]
LGTM

File: examples/platforms/simulation/system.c [hunk 4]
```diff
@@ -63,56 +64,101 @@ static void handleSignal(int aSignal)
...
 
+/**
+ * This enumeration defines the argument return values.
+ *
+ */
+enum
+{
+    OT_SIM_OPT_HELP        = 'h',
+    OT_SIM_OPT_SLEEP_TO_TX = 's',
+    OT_SIM_OPT_TIME_SPEED  = 't',
```
Comment (bukepo) [inline examples/platforms/simulation/system.c:75]: On POSIX platform, this is `s`. Is it better to use the same?

Comment (Irving-cl) [reply examples/platforms/simulation/system.c]: Updated.

File: examples/platforms/simulation/system.c [hunk 5]
```diff
@@ -63,56 +64,101 @@ static void handleSignal(int aSignal)
...
         return;
     }
 
-    if (aArgCount < 2)
+    while (true)
     {
-        fprintf(stderr, "Syntax:\n    %s [--sleep-to-tx] NodeId [TimeSpeedUpFactor]\n", aArgVector[0]);
-        exit(EXIT_FAILURE);
-    }
+        int c = getopt_long(aArgCount, aArgVector, "", long_options, NULL);
```
Comment (bukepo) [inline examples/platforms/simulation/system.c:113]:
```suggestion
        int c = getopt_long(aArgCount, aArgVector, "hts:", long_options, NULL);
```

File: examples/platforms/simulation/system.c [hunk 6]
```diff
@@ -63,56 +64,101 @@ static void handleSignal(int aSignal)
...
+    exit(aExitCode);
+}
+
 void otSysInit(int aArgCount, char *aArgVector[])
 {
     char *   endptr;
     uint32_t speedUpFactor = 1;
-    int      argIndex      = 0;
+
+    struct option long_options[] = {
```
Comment (bukepo) [inline examples/platforms/simulation/system.c:98]:
```suggestion
    static const struct option long_options[] = {
```

File: examples/platforms/simulation/system.c [hunk 7]
```diff
@@ -63,56 +64,101 @@ static void handleSignal(int aSignal)
...
+    };
 
     if (gPlatformPseudoResetWasRequested)
     {
         gPlatformPseudoResetWasRequested = false;
         return;
     }
 
-    if (aArgCount < 2)
+    while (true)
```
Comment (bukepo) [inline examples/platforms/simulation/system.c:111]: `optind` is global variable, initialize it because we use `exec` for resetting. 
```suggestion
    optind = 1;

    while (true)
```

File: examples/platforms/simulation/system.c [hunk 8]
```diff
@@ -63,56 +64,101 @@ static void handleSignal(int aSignal)
...
+ */
+enum
+{
+    OT_SIM_OPT_HELP        = 'h',
+    OT_SIM_OPT_SLEEP_TO_TX = 't',
+    OT_SIM_OPT_TIME_SPEED  = 's',
+    OT_SIM_OPT_UNKNOWN     = '?',
+};
+
+static void printUsage(const char *aProgramName, int aExitCode)
```
Comment (bukepo) [inline examples/platforms/simulation/system.c:79]:
```suggestion
static void PrintUsage(const char *aProgramName, int aExitCode)
```

**Review by zhanglongxia — APPROVED** [review]

**Review by bukepo — COMMENTED** [review]
LGTM:+1:

**Review by bukepo — APPROVED** [review]

File: examples/platforms/simulation/system.c [hunk 9]
```diff
@@ -63,56 +64,103 @@ static void handleSignal(int aSignal)
...
+    setlogmask(setlogmask(0) & LOG_UPTO(LOG_NOTICE));
 
-        if (*endptr != '\0' || speedUpFactor == 0)
-        {
-            fprintf(stderr, "Invalid value for TimerSpeedUpFactor: %s\n", aArgVector[argIndex]);
-            exit(EXIT_FAILURE);
-        }
-    }
+    signal(SIGTERM, &handleSignal);
+    signal(SIGHUP, &handleSignal);
```
Comment (jwhui) [inline examples/platforms/simulation/system.c:163]: Are these changes relevant to the `get_opt` change?

Comment (Irving-cl) [reply examples/platforms/simulation/system.c:163]: Yeah. The checking for `speedUpFactor` is moved into the parsing process above. The calling of `openlog` and `signal` are moved after the parsing process.

**Review by jwhui — APPROVED** [review]
